migrations/20200325204238-mv-data-from-accounts-to-recoverymethods.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+module.exports = {
+    up: queryInterface => {
+        return queryInterface.sequelize.transaction(transaction => {
+            return new Promise(resolve => {
+                Promise.all([
+                    queryInterface.sequelize.query(
+                        'SELECT * FROM "Accounts" WHERE "phoneNumber" IS NOT NULL',
+                        { transaction }
+                    ),
+                    queryInterface.sequelize.query(
+                        'SELECT * FROM "Accounts" WHERE "email" IS NOT NULL',
+                        { transaction }
+                    )
+                ]).then(([[phoneRecords], [emailRecords]]) => {
+                    Promise.all([
+                        phoneRecords.length && queryInterface.bulkInsert(
+                            'RecoveryMethods',
+                            phoneRecords.map(account => ({
+                                AccountId: account.id,
+                                kind: 'phone',
+                                detail: account.phoneNumber,
+                                createdAt: account.updatedAt,
+                                updatedAt:account.updatedAt,
+                            })),
+                            { transaction }
+                        ),
+                        emailRecords.length && queryInterface.bulkInsert(
+                            'RecoveryMethods',
+                            emailRecords.map(account => ({
+                                AccountId: account.id,
+                                kind: 'email',
+                                detail: account.email,
+                                createdAt: account.updatedAt,
+                                updatedAt:account.updatedAt,
+                            })),
+                            { transaction }
+                        )
+                    ]).then(resolve).catch(err => {
+                        console.error(err);
+                        return transaction.rollback();
+                    });
+                }).catch(err => {
+                    console.error(err);
+                    return transaction.rollback();
+                });
+            });
+        });
+    },
+
+    down: queryInterface => {
```

Comment:
just FYI: I don't think there is any context where we currently use "down" migrations, so you might save time skipping them

Reply:
If we need to roll back a change we would need to roll back migrations, correct?

Reply:
in theory – yes. In practice I had to resort to manual rollbacks most of the time I was running Sequelize in production (few years). Down migrations don't get tested, state is inconsistent because of something unforeseen, etc.